apps/src/netsim/NetSimRemoteNodeSelectionPanel.js

```diff
@@ -0,0 +1,200 @@
+
+  // Initial render
+  NetSimPanel.call(this, rootDiv, {
+    className: 'netsim-lobby-panel',
+    panelTitle: i18n.connectToANode(),
+    canMinimize: false
+  });
+};
+NetSimRemoteNodeSelectionPanel.inherits(NetSimPanel);
+
+/**
+ * Recreate markup within panel body.
+ */
+NetSimRemoteNodeSelectionPanel.prototype.render = function () {
+  // Create boilerplate panel markup
+  NetSimRemoteNodeSelectionPanel.superPrototype.render.call(this);
+
+  // Add our own content markup
+  var newMarkup = $(markup({
+    level: this.levelConfig_,
+    nodesOnShard: this.nodesOnShard_,
+    nodesRequestingConnection: this.nodesRequestingConnection_,
+    selectedNode: this.selectedNode_,
+    remoteNode: this.remoteNode_,
+    canConnectToNode: this.canConnectToNode_.bind(this),
+    isMyNode: this.isMyNode_.bind(this)
+  }));
+  this.getBody().html(newMarkup);
+
+  this.addRouterButton_ = this.getBody().find('#netsim_lobby_add_router');
+  this.addRouterButton_.click(this.addRouterCallback_);
+
+  this.connectButton_ = this.getBody().find('#netsim_lobby_connect');
+  this.connectButton_.click(this.connectButtonCallback_);
+
+  this.cancelButton_ = this.getBody().find('#netsim_lobby_cancel');
+  this.cancelButton_.click(this.cancelButtonCallback_);
+
+  this.getBody().find('.selectable-row').click(this.onRowClick_.bind(this));
+};
+
+/**
+ * @param {Event} jQueryEvent
+ * @private
+ */
+NetSimRemoteNodeSelectionPanel.prototype.onRowClick_ = function (jQueryEvent) {
+  var target = $(jQueryEvent.target);
+  var nodeID = target.data('nodeId');
+  var clickedNode = _.find(this.nodesOnShard_, function (node) {
+    return node.entityID === nodeID;
+  });
+
+  // Don't even allow clicking on nodes we can't connect to.
+  if (!clickedNode || !this.canConnectToNode_(clickedNode)) {
+    return;
+  }
+
+  // If the selected node was clicked, we want to deselect.
+  if (this.selectedNode_ && this.selectedNode_.entityID === clickedNode.entityID) {
+    this.selectNodeCallback_(null);
+  } else {
+    this.selectNodeCallback_(clickedNode);
+  }
+};
+
+/**
+ * @param {NetSimNode} node
+ * @returns {boolean}
+ * @private
+ */
+NetSimRemoteNodeSelectionPanel.prototype.isMyNode_ = function (node) {
+  return this.myNodeID_ === node.entityID;
+};
+
+/**
+ * Check whether the level configuration allows connections to the specified
+ * node.
+ * @param {NetSimNode} connectionTarget
+ * @returns {boolean} whether connection to the target is allowed
+ * @private
+ */
+NetSimRemoteNodeSelectionPanel.prototype.canConnectToNode_ = function (connectionTarget) {
+  // Can't connect to own node
+  if (this.isMyNode_(connectionTarget)) {
+    return false;
+  }
+
+  // Permissible connection limited by level configuration
+  return (
+      this.levelConfig_.canConnectToClients &&
+      connectionTarget.getNodeType() === NodeType.CLIENT
+      ) || (
+      this.levelConfig_.canConnectToRouters &&
+      connectionTarget.getNodeType() === NodeType.ROUTER );
```

Comment:
Indentation here seems a little hard to parse, could throw this `);` on a newline, & maybe de-indent the `) || (` one? Alternatively could extract variable to name one or more of the inner boolean statements